### PR TITLE
Add more how-to docs

### PR DESCRIPTION
**Pull request by brent-hoover**
Resolves: https://github.com/reactioncommerce/docs/issues/34

Adds from old docs

How to:

Store custom fields on an order
Create an inventory plugin
Create a GraphQL query
Create a GraphQL mutation
Extend GraphQL to add field
Extend GraphQL with remote schema
Emit and Listen for Events (significantly rewritten)
Run Functions on Startup
Add Collections from a plugin
Write and run database migrations
Misc How to

Core:
Developing with GraphQL

New doc created from scratch:
How to extend a schema

### CODE REVIEW

File: guides/developers-guide/core/developing-graphql.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+The plugin function:
+- Lives in `queries` or `mutations` in a plugin folder
+- Is available on the GraphQL context in `context.queries` or `context.mutations`, and as such can be called by code elsewhere in the app
+- Returns a Promise (is async)
+- Does all permission checks
+- May throw a `ReactionError` if anything goes wrong
+- Performs the actual database mutations or queries
+
+TIP: If you’re confused about where to draw the line, imagine what would have to change if we decided to add a REST API. All of that stuff goes in the resolver, while everything that would be shared between GraphQL and REST goes in the plugin function.
```
Comment (delagroove) [inline guides/developers-guide/core/developing-graphql.md]: good tip!

File: guides/developers-guide/core/developing-graphql.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+## The Endpoint
+
+The GraphQL server and `/graphql` endpoint is configured and returned by the `createApolloServer` function, which is called from the `ReactionAPI` class instance.
+
+`createApolloServer` does pretty standard configuration of an Express app using `apollo-server-express`. The main things it does are:
+- Checks the identity token using Express middleware
```
Comment (delagroove) [inline guides/developers-guide/core/developing-graphql.md]: @zenweasel is this still happening in 4.0?

Comment (delagroove) [reply guides/developers-guide/core/developing-graphql.md]: just making sure it does not refer to something from the identity service

File: guides/developers-guide/core/developing-graphql.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+All GraphQL resolvers receive a [context](https://www.apollographql.com/docs/apollo-server/data/resolvers/#the-context-argument) object as their third argument. The base context is built within the `ReactionAPI` constructor, and additional request-specific properties (like `accountId` and `userHasPermission`) are added to it in `buildContext.js`.
+
+In Jest tests, you can get a mock context object with mock functions on it:
+
+```js
+import mockContext from "/imports/test-utils/helpers/mockContext";
+```
+
+Here’s what's on the context object:
+- Queries registered by plugins: `context.queries[queryFunctionName]`
```
Comment (MohanNarayana) [inline guides/developers-guide/core/developing-graphql.md]: Can we use format 'context.queries.\<queryFunctionName>'
Same for line 75 and 83

**Comment by Akarshit** [comment]
We should also include the latest guide we have from postlight about writing documentation for Mailchimp. Not a blocker for this issue but definitely something we should do. 
@delagroove Can you share that with Brent?

File: guides/developers-guide/core/developing-graphql.md (hunk 1 above)
Comment (Akarshit) [reply guides/developers-guide/core/developing-graphql.md]: This was confusing for me. Maybe a table for side by side comparison and an example would work best here.

File: guides/developers-guide/core/developing-graphql.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+- To check permissions and throw error: `context.checkPermissions(role, shopId)`
+- MongoDB collections: `context.collections[CollectionName]`
+- The `ReactionAPI` instance: `context.app`
+- App events object:
+    - To emit: `context.appEvents.emit`
+    - To listen: `context.appEvents.on`
+- To retrieve all functions registered as a specific type of function: `context.getFunctionsOfType(type)`
+- The app root URL: `context.rootUrl`
+- To convert a relative URL to absolute (prefix with the root URL): `context.getAbsoluteUrl(path)`
+
```
Comment (Akarshit) [inline guides/developers-guide/core/developing-graphql.md:92]: Things I found missing: 
1. `appVersion`
2. `auth`
3. `getInternalContext()`
4. `mutations`
5. `queries`
6. `pubSub`
7. All custom fields added by the plugin through `contextAdditions`
I also support moving this out to a separate doc and linking it here.

File: guides/developers-guide/core/developing-graphql.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+The GraphQL resolver functions are the place where ID encoding and decoding happens. They then call out to plugin functions that deal exclusively with internal IDs. Any IDs returned by such functions must also be transformed before returning them, although this typically and preferably happens in a type resolver.
+
+## Using MongoDB in GraphQL
+
+The source-of-truth database for most data used by GraphQL resolvers is currently MongoDB. The MongoDB Node driver collections API is documented here: http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html
+
+All collections registered by any plugin are available on the `context` object as `context.collections`.
+
+### Checking whether an operation was successful
```
Comment (Akarshit) [inline guides/developers-guide/core/developing-graphql.md:117]: Do we need this documentation? Seems not related to Reaction to me.

File: guides/developers-guide/core/developing-graphql.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+{
+  order {
+    shop {
+      _id
+    }
+  }
+}
+```
+
+Normally the `shop` relationship would result in a database query, but if `order` already has a `shopId` property, we can actually skip the database lookup because the client has requested only the `_id` property. There is a utility function that helps with this: `optimizeIdOnly`. Check out the `Query.viewer` resolver for an example of how to use it.
```
Comment (Akarshit) [inline guides/developers-guide/core/developing-graphql.md:162]: Maybe we should add link to the `Query.viewer` resolver here.

File: how-tos/add-collections-from-plugin.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    // other props
+  });
+}
+```
+
+The `collections` object key is where you will access this collection on `context.collections`, and `name` is the collection name in MongoDB. We recommend you make these the same if you can.
+
+The example above will make `context.collections.MyCustomCollection` available in all query and mutation functions, and all functions that receive `context`, such as startup functions. Note that usually MongoDB will not actually create the collection until the first time you insert into it.
+
+If you need to ensure indexes on any fields in your collection, see the next section.
```
Comment (Akarshit) [inline how-tos/add-collections-from-plugin.md]: We can delete this line, the next line is exactly the same thing.

File: how-tos/add-extra-data-to-a-product.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+- Create a GraphQL mutation for setting your custom property value
+- Create a React component that allows an operator to set the custom property value, and wire it up to your mutation, or set your property in some other way
+
+### Extend database schemas
+
+To extend any database schema, you just need a file that is imported into server code. We recommend using a file named `simpleSchemas.js` in your plugin, and then importing that file in your plugin's `index.js`.
+
+Refer to [SimpleSchema docs](https://github.com/aldeed/simple-schema-js#schema-rules) for more information about the object you pass to `extend`.
+
+```js
```
Comment (Akarshit) [inline how-tos/add-extra-data-to-a-product.md:26]: We should say where exactly to do this. Like in the register function, startup function etc.

File: how-tos/add-extra-data-to-a-product.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+};
+
+// Extend the Product database schema, if your custom property will be on products
+Product.extend(schemaExtension);
+
+// Extend the Variant database schema, if your custom property will be on variants
+ProductVariant.extend(schemaExtension);
+
+// Extend the CatalogProduct database schema, if your custom property will be on products
+CatalogProduct.extend(schemaExtension);
```
Comment (Akarshit) [inline how-tos/add-extra-data-to-a-product.md:43]: Maybe we can move this to line 38 so that they are together.

File: how-tos/add-extra-data-to-a-product.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+  await app.registerPlugin({
+    functionsByType: {
+      publishProductToCatalog: [publishProductToCatalog]
+    },
+    // other props
+  });
+}
+```
+
+Where the `publishProductToCatalog` function looks something like this:
```
Comment (Akarshit) [inline how-tos/add-extra-data-to-a-product.md]: What about - "Where the `publishProductToCatalog` function adds the required fields to `catalogProduct`. For example:"

File: how-tos/add-extra-data-to-a-product.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+```js
+export default function publishProductToCatalog(catalogProduct, { context, product, shop, variants }) {
+  catalogProduct.myProperty = product.myProperty;
+  // Also set on each catalogProduct.variants if necessary
+}
+```
+
+### Create a GraphQL mutation for setting your custom property value
+
+Refer to [How To: Create a new GraphQL mutation](./create-graphql-mutation.md). After mutating the `Product` document in your function, you must also call `context.mutations.hashProduct(productId, context.collections, false)`. This will update the current product hash, causing the operator UI to indicate that there are changes needing publishing.
```
Comment (Akarshit) [inline how-tos/add-extra-data-to-a-product.md]: Just to follow specifications: `context.mutations.hashProduct(<productId>, context.collections, false)`

**Comment by Akarshit** [comment]
Still going through the other files. This is awesome work!

**Comment by delagroove** [comment]
@Akarshit the Postlight documentation guide has been shared in notion for @zenweasel to check

File: how-tos/create-and-run-migrations.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
+# How to create and run migrations
```
Comment (MohanNarayana) [inline how-tos/create-and-run-migrations.md:1]: I had written [this doc](https://github.com/reactioncommerce/docs/blob/trunk/guides/API-plugin-package-migration.md) a few months ago with more detailed steps. It could add to this doc.

File: guides/developers-guide/core/developing-graphql.md (hunk 4 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md:92]: @Akarshit Don't you have already have an issue to create a "what is the `context` object? Does it make sense to put that in our TODO list for stuff we should work on now?

File: guides/developers-guide/core/developing-graphql.md (hunk 1 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md]: Just changed the tip. See if that makes more sense

File: guides/developers-guide/core/developing-graphql.md (hunk 5 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md:117]: Removed

File: guides/developers-guide/core/developing-graphql.md (hunk 6 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md:162]: Where is that located?

File: how-tos/add-extra-data-to-a-product.md (hunk 8 above)
Comment (brent-hoover) [reply how-tos/add-extra-data-to-a-product.md:26]: This is referenced in the "Requisite Reading" doc on how to extend a schema. We shouldn't repeat it here.

File: how-tos/add-extra-data-to-a-product.md (hunk 9 above)
Comment (brent-hoover) [reply how-tos/add-extra-data-to-a-product.md:43]: I think we are grouping Product extensions together, and then Catalog together. I think keeping them apart makes sense.

File: how-tos/add-extra-data-to-a-product.md (hunk 10 above)
Comment (brent-hoover) [reply how-tos/add-extra-data-to-a-product.md]: Updated with change

File: how-tos/add-extra-data-to-a-product.md (hunk 11 above)
Comment (brent-hoover) [reply how-tos/add-extra-data-to-a-product.md]: Updated with change

File: how-tos/create-and-run-migrations.md (hunk 12 above)
Comment (brent-hoover) [reply how-tos/create-and-run-migrations.md:1]: I stole some stuff from that doc and fixed a couple of issues with it. Take another read and see if it makes sense/is accurate. Generally I think this whole migration process needs a lot of work but we can go back to that.

File: how-tos/create-and-run-migrations.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+4. Create `index.js` which exports the namespace and migration script.
+5. Add `export { default as migrations } from "./migrations/index.js";` in your plugin entry point file i.e, `<api-plugin-package>/index.js`.
+6. Add the latest version of the `@reactioncommerce/db-version-check` NPM package as a dependency using `npm install @reactioncommerce/db-version-check@latest`.
+7. Add and register a `preStartup` function in `src/index.js`. In it, call `doesDatabaseVersionMatch` to prevent API startup if the data isn't compatible with the code. The preStartup.js is responsible for identifying version mismatch. It throws an error if a migration is required and prevents api from running.
+
+### In api-migrations
+1. Add the new track to `api-migrations/migrator.config.js` following the existing syntax. The app looks at this file to pick the desired version and checks against the migrations table in the DB to identify the current version.
+2. Install the plugin package using `npm install @reactioncommerce/<api-plugin-package>
```
Comment (Akarshit) [inline how-tos/create-and-run-migrations.md]: MIssing end of "`"

File: how-tos/create-graphql-mutation.md [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+import mockContext from "/imports/test-utils/helpers/mockContext";
+import createSomething from "./createSomething";
+
+test("expect to return a Promise that resolves to null", async () => {
+  const result = await createSomething(mockContext);
+  expect(result).toEqual(null);
+});
+```
+
+This of course should be updated with tests that are appropriate for whatever your mutation does.
```
Comment (Akarshit) [inline how-tos/create-graphql-mutation.md]: Maybe we can skip this line.

File: how-tos/create-graphql-mutation.md [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+  }, context);
+
+  expect(result).toEqual({
+    renameMe: fakeResult,
+    clientMutationId: "clientMutationId"
+  });
+});
+```
+
+This of course should be updated with tests that are appropriate for whatever your mutation resolver does. For example, verify that all ID and schema transformations happen.
```
Comment (Akarshit) [inline how-tos/create-graphql-mutation.md:220]: Skip this too?

File: how-tos/run-function-on-startup.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    label: "Shipping",
+    name: "reaction-shipping",
+    functionsByType: {
+      startup: [startup]
+    }
+    // other props
+  });
+}
+```
+
```
Comment (Akarshit) [inline how-tos/run-function-on-startup.md:30]: Extra line?

File: how-tos/emit-and-listen-to-events.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+## Emit an app event
+
+Emit app events in API code using `appEvents.emit`. The `emit` function takes at least two parameters: the name of the event as a string and the `payload` of functions as an object.
+
+### Function parameters and options
+
+- *Event name*: The first argument is the event name, as a string. There is currently no limit to what event name you can emit, but generally try to follow established patterns for naming. See events table below.
+- *Payload*: The second argument, the `payload`, should always be an object, and not an entity object. Rather than passing `order` in directly, pass it in an object: `{ order }`, so that more fields can be added or removed from the payload more easily.
+- *Option arguments*: The last argument is an array of arguments to pass to each function from `payload`.
+- *Using `await`*: Using the method with `await` will not resolve until all registered handler methods have resolved.
```
Comment (Akarshit) [inline how-tos/emit-and-listen-to-events.md]: This should be in a `NOTE`

File: how-tos/emit-and-listen-to-events.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+const { appEvents } = context;
+```
+
+## Emit an app event
+
+Emit app events in API code using `appEvents.emit`. The `emit` function takes at least two parameters: the name of the event as a string and the `payload` of functions as an object.
+
+### Function parameters and options
+
+- *Event name*: The first argument is the event name, as a string. There is currently no limit to what event name you can emit, but generally try to follow established patterns for naming. See events table below.
```
Comment (Akarshit) [inline how-tos/emit-and-listen-to-events.md]: Didn't find the event table.

File: how-tos/extend-an-existing-schema.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+  purchaseOrderNumber: String
+});
+
+export default function extendSchemas(context) {
+  const { simpleSchemas: { Order } } = context;
+  Order.extend(purchaseOrderSchema);
+}
+ ```
+
+Since `context` here serves as a giant Singleton that is shared throughout the entire app, your now mutated schema is not available everywhere.
```
Comment (Akarshit) [inline how-tos/extend-an-existing-schema.md]: Should be "is available everywhere"

File: how-tos/how-do-i-misc.md [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+```js
+await context.mutations.createNotification(context, {
+  accountId,
+  type: "orderCanceled",
+  url
+});
+```
+
+
+Each item in the `indexes` array is an array of arguments that will be passed to the Mongo `createIndex` function. The `background` option is always set to `true` so you need not include that.
```
Comment (Akarshit) [inline how-tos/how-do-i-misc.md]: Seems like something is missing here.

File: how-tos/how-do-i-misc.md [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+
+## Loop over async function results
+
+Often you have a list of functions that return a Promise, and you need to loop through the list and call each function. The recommended way to do this depends on whether the functions are expecting to be called in series or can be safely called in parallel.
+
+For performance reasons, you should call them in parallel if you can. To do so, use `map` and await `Promise.all`.
+
+```js
+const promisedResults = listOfFunctions.map((func) => func());
+const results = await Promise.all(promisedResults);
```
Comment (Akarshit) [inline how-tos/how-do-i-misc.md:53]: Can't we just do `const results = await Promise.all(listOfFunctions);`?

File: guides/developers-guide/core/developing-graphql.md [hunk 22]
```diff
@@ -0,0 +1,182 @@
...
+## Extending and Modifying the GraphQL API
+
+The GraphQL schema is written in multiple `.graphql` files, which contain type definitions in the GraphQL schema language. These files live in the plugins to which they relate, in a `schemas` folder. Refer to one of the How To articles:
+
+- [How To: Create a new GraphQL query](../../../how-tos/create-graphql-query.md)
+- [How To: Create a new GraphQL mutation](../../../how-tos/create-graphql-mutation.md)
+
+## Documenting a GraphQL Schema
+
+Every type, input type, enum, query, mutation, and field must have documentation. Add a description using a string literal immediately above the thing you are documenting in the `.graphql` file. <strong>This is the official API documentation, so take the time to make it clear, well-formatted, and with no spelling or grammar errors.</strong>
```
Comment (delagroove) [inline guides/developers-guide/core/developing-graphql.md:12]: @zenweasel is there a Markdown way to use bold? don't think <strong> works as intended.

Comment (delagroove) [reply guides/developers-guide/core/developing-graphql.md:12]: bold is better with two asterisks opening and closing: **bold text**
__BOLD__
underscores should work too

File: guides/developers-guide/core/developing-graphql.md [hunk 23]
```diff
@@ -0,0 +1,182 @@
...
+- Checks the identity token using Express middleware
+- Builds the `context` object that’s available in all resolver functions. See [The Reaction GraphQL Context](#the-reaction-graphql-context)
+- Formats the `errors` array that is returned to clients, to make errors as helpful as possible
+- Provides the merged GraphQL schema
+- Sets the path as `/graphql` and exposes a GraphQL Playground for GET requests on `/graphql`
+
+## The Reaction GraphQL Context
+
+All GraphQL resolvers receive a [context](https://www.apollographql.com/docs/apollo-server/data/resolvers/#the-context-argument) object as their third argument. The base context is built within the `ReactionAPI` constructor, and additional request-specific properties (like `accountId` and `userHasPermission`) are added to it in `buildContext.js`.
+
```
Comment (delagroove) [inline guides/developers-guide/core/developing-graphql.md:67]: @Akarshit if this part is also related with `context` object make sure you link them

File: guides/developers-guide/core/developing-graphql.md [hunk 24]
```diff
@@ -0,0 +1,182 @@
...
+In particular, note that "it is not intended to be human‐readable". You should never display a field of type `ID` anywhere. They are only for references. If your data comes over from another system and has IDs with some meaning, then you should also store them on a different field where the raw value will not be obfuscated by the GraphQL layer.
+
+Note also that the server specification does not necessarily care whether an ID is globally unique. However, we intend compatibility with both Relay and Apollo for client-side frameworks, and [the Relay specification](https://facebook.github.io/relay/graphql/objectidentification.htm#sec-Node-Interface) does have a requirement here:
+
+> This `id` should be a globally unique identifier for this object, and given just this `id`, the server should be able to refetch the object.
+
+Additionally, the [Apollo caching docs](https://www.apollographql.com/docs/react/advanced/caching/#normalization) have this to say:
```
Comment (delagroove) [inline guides/developers-guide/core/developing-graphql.md]: there's no normalization section on caching, found https://www.apollographql.com/docs/react/caching/cache-configuration/#disabling-normalization but don't think is the one. please update

File: how-tos/create-and-run-migrations.md (hunk 13 above)
Comment (delagroove) [inline how-tos/create-and-run-migrations.md]: need to close with `

File: guides/developers-guide/core/developing-graphql.md (hunk 22 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md:12]: Depending on what flavor of Markdown you are using a limited amount of HTML tags work as you can see [here](https://raw.githubusercontent.com/reactioncommerce/reaction/trunk/README.md) but we should probably use the more standard double asterisks.

Pretty sure underscores give you _italics_

File: guides/developers-guide/core/developing-graphql.md (hunk 2 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md]: I still think we check for a token but where it comes from has changed? @Akarshit Can you confirm? Good call out

Comment (Akarshit) [reply guides/developers-guide/core/developing-graphql.md]: It's still the same. Maybe we can just replace identity token with "access token"

File: docs/jsdoc-style-guide.md [hunk 25]
```diff
@@ -0,0 +1,38 @@
+---
+id: jsdoc-style-guide
```
Comment (MohanNarayana) [inline docs/jsdoc-style-guide.md]: Is this id, title table needed? I have removed the table in dev concepts docs and only kept the title. We can either have it in all docs or just have title, for consistency.

File: how-tos/emit-and-listen-to-events.md [hunk 26]
```diff
@@ -0,0 +1,66 @@
+# How do I: Emit and Listen for AppEvents
+
+## Background:
+
+AppEvents is a simple interface that allows you to build a more event-drive approach. This allows one part of the app to announce that something important has happened, and other parts of the app
+to "react" without the emitter needing to know anything about the listeners. We call sending events "emitting" and listeners "handlers". A simple example of where RC uses appEvents is that when
```
Comment (MohanNarayana) [inline how-tos/emit-and-listen-to-events.md]: RC should be OC

File: how-tos/extend-an-existing-schema.md [hunk 27]
```diff
@@ -0,0 +1,66 @@
+# How to extend a Simple Schema
+
+## Background
+
+While Reaction uses a (mostly) schemaless database, we do value data validation and do that by using the package [SimpleSchema](https://www.npmjs.com/package/simpl-schema) which provides
```
Comment (MohanNarayana) [inline how-tos/extend-an-existing-schema.md]: Reaction should be Open Commerce

File: how-tos/extend-graphql-with-remote-schema.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
+# How To: Extend GraphQL with Remote Schema Delegation
+
+If you have an external service providing a GraphQL interface and you would like to make it available via the main Reaction GraphQL API, here's how to do it. For this example, we'll use the [public sample Pokemon GraphQL API](https://github.com/lucasbento/graphql-pokemon) from `https://graphql-pokemon.now.sh`.
```
Comment (MohanNarayana) [inline how-tos/extend-graphql-with-remote-schema.md]: Reaction should be Open commerce in line 3 and 21

File: how-tos/run-function-on-startup.md [hunk 29]
```diff
@@ -0,0 +1,30 @@
+# How to Run plugin code on app startup
+Copy the following into a startup.js file in the plugin folder:
```
Comment (Akarshit) [inline how-tos/run-function-on-startup.md:2]: We should also mention about pre-startup, post-startup phases here.

File: guides/developers-guide/core/developing-graphql.md (hunk 6 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md:162]: @Akarshit second request for clarification here

File: guides/developers-guide/core/developing-graphql.md (hunk 24 above)
Comment (brent-hoover) [reply guides/developers-guide/core/developing-graphql.md]: Good catch. I'm just removing it for now and will re-review it later with Eric

File: how-tos/create-graphql-mutation.md (hunk 15 above)
Comment (brent-hoover) [reply how-tos/create-graphql-mutation.md:220]: Why? Why does it hurt to specify that you should write tests?

File: how-tos/how-do-i-misc.md (hunk 21 above)
Comment (brent-hoover) [reply how-tos/how-do-i-misc.md:53]: I don't think so, that just gives you an array of functions, not an array of promises. You need to execute the function with the arguments to get a promise.

File: docs/jsdoc-style-guide.md (hunk 25 above)
Comment (brent-hoover) [reply docs/jsdoc-style-guide.md]: Yes, you're correct, that stuff should be removed

File: how-tos/run-function-on-startup.md (hunk 29 above)
Comment (brent-hoover) [reply how-tos/run-function-on-startup.md:2]: Is there an after startup phase? I am familiar with `preStartup` but what's the other one? I don't see anything about that in `api-core`?

**Comment by brent-hoover** [comment]
@MohanNarayana @delagroove @Akarshit Changes from C/R made. Please review

**Review by delagroove — APPROVED** [review]

File: how-tos/run-function-on-startup.md (hunk 29 above)
Comment (Akarshit) [reply how-tos/run-function-on-startup.md:2]: No there isn't just the preStartup.

File: how-tos/create-graphql-mutation.md (hunk 15 above)
Comment (Akarshit) [reply how-tos/create-graphql-mutation.md:220]: Mostly because I feel that we don't need to tell people that. Giving a example implies that they can modify it according to their needs. Anyway not a big deal.

File: guides/developers-guide/core/developing-graphql.md (hunk 6 above)
Comment (Akarshit) [reply guides/developers-guide/core/developing-graphql.md:162]: Here is the link: https://github.com/reactioncommerce/api-plugin-accounts/blob/9306d27721a8fe5edf1793d8def4999d9395ddbc/src/resolvers/Query/viewer.js#L17

File: how-tos/run-function-on-startup.md (hunk 29 above)
Comment (Akarshit) [reply how-tos/run-function-on-startup.md:2]: But I still think we should mention about `preStartup`, `startup` & `shutdown` here. This seems out of scope for the title of the doc. So maybe this could be a "run function lifecycle" doc?

File: how-tos/emit-and-listen-to-events.md [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+  if (emittedBy === EMITTED_BY_NAME) return; // short circuit infinite loops
+
+  appEvents.emit("afterCartUpdate", { cart: updatedCart, updatedBy: userId }, { emittedBy: EMITTED_BY_NAME });
+});
+```
+
+## Events
+
+Developers are encouraged to add AppEvent emitters to their code whenever they create a serious event, allowing custom plugin authors to use this for extension
+
```
Comment (MohanNarayana) [inline how-tos/emit-and-listen-to-events.md:67]: We can add [this](https://github.com/reactioncommerce/reaction-docs/blob/c05fe9d21ab2af3103437a6f971531120d4513d3/public-docs/appevent-hooks.md?plain=1#L55) table here (if needed) and get rid of the dev concepts events doc

Comment (brent-hoover) [reply how-tos/emit-and-listen-to-events.md:67]: We absolutely do not want that table as it is very much out of date.

File: how-tos/run-function-on-startup.md (hunk 29 above)
Comment (brent-hoover) [reply how-tos/run-function-on-startup.md:2]: I think mentioning only startup in a doc entitled "running things on startup" makes a lot of sense

**Comment by brent-hoover** [comment]
Closing in favor of creating smaller PR's
